src/packages/admin-ui-components/src/detail-panel/component.tsx

```diff
@@ -348,32 +362,71 @@ export const DetailPanel = () => {
 		window.sessionStorage.removeItem(persistName);
 	};
 
+	const handleConfirmLeave = () => {
+		if (!modalRedirectUrl) return;
+
+		navigate(modalRedirectUrl, { replace: true });
+	};
+
+	const handleCancelLeave = () => {
+		// Close the unsaved changes modal via clearing the modalRedirectUrl
+		setSearchParams({ modalRedirectUrl: '' });
+	};
+
 	return (
-		<Modal
-			isOpen
-			onRequestClose={closeModal}
-			shouldCloseOnEsc
-			shouldCloseOnOverlayClick
-			className={open ? classnames(styles.detailContainer, styles.slideIn) : styles.detailContainer}
-			title={selectedEntity.name}
-			modalContent={
-				<>
-					{loading ? (
-						<Spinner />
-					) : error ? (
-						<p>Failed to load entity.</p>
-					) : (
-						<DetailForm
-							initialValues={initialValues}
-							detailFields={formFields}
-							onCancel={closeModal}
-							onSubmit={handleOnSubmit}
-							persistName={persistName}
-							isReadOnly={selectedEntity.attributes.isReadOnly}
-						/>
-					)}
-				</>
-			}
-		/>
+		<>
+			<Modal
+				key={selectedEntity.name}
+				isOpen
+				onRequestClose={closeModal}
+				shouldCloseOnEsc
+				shouldCloseOnOverlayClick
+				className={
+					open ? classnames(styles.detailContainer, styles.slideIn) : styles.detailContainer
+				}
+				title={selectedEntity.name}
+				modalContent={
+					<>
+						{loading ? (
+							<Spinner />
+						) : error ? (
+							<p>Failed to load entity.</p>
+						) : (
+							<>
+								<DetailForm
+									initialValues={initialValues}
+									detailFields={formFields}
+									onCancel={closeModal}
+									onSubmit={handleOnSubmit}
+									persistName={persistName}
+									isReadOnly={selectedEntity.attributes.isReadOnly}
+								/>
+							</>
+						)}
+					</>
+				}
+			/>
+			<Modal
+				isOpen={!!modalRedirectUrl}
+				hideCloseX
+				modalContent={
+					<>
+						<div className={styles.unsavedChangesTitle}>You have unsaved changes</div>
+						<div className={styles.unsavedChangesContent}>
+							Are you sure you want to leave this entity? Your changes will not be saved.
+						</div>
+						<div className={styles.unsavedChangesButtonRow}>
+							<Button className={styles.unsavedChangesButton} onClick={handleCancelLeave}>
```

Comment:
Can you get latest from main? I had updated the style for reset type. Can you use reset type button? `<Button type="reset" onClick={handleCancelLeave}>
								Cancel
							</Button>
							<Button type="reset" onClick={handleConfirmLeave}>
								Discard Changes
							</Button>`

Reply:
Oh ya will do, thanks!

Reply:
ah no these don't match my modal designs... thank you though.